src/pipelinedefinition.cpp

```diff
@@ -31,10 +34,51 @@ Status toNodeKind(const std::string& str, NodeKind& nodeKind) {
     return StatusCode::PIPELINE_NODE_WRONG_KIND_CONFIGURATION;
 }
 
+Status PipelineDefinition::validate(ModelManager& manager) {
+    struct ValidationResultNotifier {
+        ValidationResultNotifier() {}
+        ~ValidationResultNotifier() {
+            if (passed) {
+                // status.notifyValidationPassed();
+            } else {
+                // status.notifyValidationFailed();
+            }
+        }
+        bool passed = false;
```

Comment:
Maybe instead of this member variable just rely on validationStatus instance?
Instead of:
`if (passed)`
call
`if (validationStatus.ok())`?

Reply:
We could add pipeline definition status and validation status as parameter to ValidationResultNotifier constructor. Looks like these are the only members needed for notifier.

Reply:
This part will need to be checked when status will be added